lib/group/group.models.js

```diff
@@ -0,0 +1,38 @@
+const Joi = require('joi');
+
+module.exports = {
+  db: (sequelize, Sequelize) => {
+    const Group = sequelize.define('Group', {
+      id: {
+        type: Sequelize.UUID,
+        defaultValue: Sequelize.UUIDV4,
+        primaryKey: true
+      },
+      title: Sequelize.STRING,
+      short_description: Sequelize.STRING,
+      description: Sequelize.TEXT,
+    });
+
+
+    Group.associate = function (models) {
+      models.Group.belongsTo(models.Account, {as: 'Admin'});
+      models.Group.hasMany(models.Post);
+    };
+
+    return Group;
+  },
+  id: Joi.object().keys({
+    // id: Joi.string().guid().required(),
+    id: Joi.string().required(),
+  }),
+  api: Joi.object().keys({
+      title: Joi.string(),
+      short_description: Joi.string(),
+      description: Joi.string(),
+  }),
+  apiFilterQuery: Joi.object({
+    search: Joi.string().optional(),
+    title: Joi.string().optional(),
+    // If search is used, other params shouldn't be
+  }).without('search', ['title']),
```

Comment:
I'm guessing this is just a copy-over from other code, but if you only query one field in your db search (say, title), you probably don't need a separate search query param. I had this because in housingdb core a user can filter by 3 or 4 fields, OR enter info in the search query and let the system try it against a couple fields instead (if the user doesn't know which field would return the result they are looking for). That said, this could be valid if 'title' only looks in the 'title' field in the db, and 'search' would go through 'title' or 'description'. So, not necessarily anything needs changed, but this also wouldn't really be needed until you implement search.

Reply:
Thank you so much for your input and explaining what the code is doing! Do you think it would be better if i remove `.without('search', ['title'])` and have the code be: 

```
apiFilterQuery: Joi.object({
    search: Joi.string().optional(),
    title: Joi.string().optional(),
    // If search is used, other params shouldn't be
  })
```

Reply:
I'd suggest just commenting it out and commenting out its use in .routes.js until you all actually need to implement it. You only need a search field if you are doing a generalized search of several fields, vs specific search using individual fields (Using search for 'xyz in a OR b OR c' vs individual fields for 'x in a AND y in b AND z in c'). And I think in the meeting I suggested just ditching filtering altogether until after MVP is reached.